src/main/java/org/broadinstitute/consent/http/service/DatasetService.java

```diff
@@ -312,27 +307,34 @@ public DataSetDTO getDatasetDTO(Integer datasetId) {
 
     public void deleteDataset(Integer datasetId) {
         List<Integer> idList = Collections.singletonList(datasetId);
-        dataSetDAO.deleteDataSetsProperties(idList);
-        dataSetDAO.deleteDataSets(idList);
+        datasetDAO.deleteDataSetsProperties(idList);
+        datasetDAO.deleteDataSets(idList);
     }
 
-    public void deleteDataset(Integer datasetId, Integer userId) throws IllegalStateException {
-        try {
-            DataSet dataset = dataSetDAO.findDataSetById(datasetId);
-            if (Objects.nonNull(dataset)) {
-                DataSetAudit dsAudit = new DataSetAudit(datasetId, dataset.getObjectId(), dataset.getName(), new Date(), true, userId, DELETE);
-                dataSetAuditDAO.insertDataSetAudit(dsAudit);
-                datasetAssociationDAO.delete(datasetId);
-                dataSetDAO.deleteDatasetPropertiesByDatasetId(datasetId);
-                if (StringUtils.isNotEmpty(dataset.getObjectId())) {
-                    dataSetDAO.logicalDatasetDelete(datasetId);
-                } else {
-                    consentDAO.deleteAssociationsByDataSetId(datasetId);
-                    dataSetDAO.deleteDatasetById(datasetId);
-                }
+    public void deleteDataset(Integer datasetId, Integer userId) throws Exception {
+        DataSet dataset = datasetDAO.findDataSetById(datasetId);
+        if (Objects.nonNull(dataset)) {
+            // Some legacy dataset names can be null
+            String dsAuditName = Objects.nonNull(dataset.getName()) ? dataset.getName() : dataset.getDatasetIdentifier();
+            DataSetAudit dsAudit = new DataSetAudit(datasetId, dataset.getObjectId(), dsAuditName, new Date(), dataset.getActive(), userId, Actions.DELETE.getValue().toUpperCase());
+            try {
+                datasetDAO.inTransaction(h -> {
+                    try {
+                        h.insertDataSetAudit(dsAudit);
+                        h.deleteUserAssociationsByDatasetId(datasetId);
+                        h.deleteDatasetPropertiesByDatasetId(datasetId);
+                        h.deleteConsentAssociationsByDataSetId(datasetId);
+                        h.deleteDatasetById(datasetId);
+                    } catch (Exception e) {
+                        h.rollback();
+                        throw e;
+                    }
+                    return true;
+                });
+            } catch (Exception e) {
+                logger.error(e.getMessage());
+                throw e;
             }
-        } catch (Exception e) {
-            throw new IllegalStateException(e.getMessage());
         }
     }
```

Comment:
This is the meat of the PR. Previous delete statements were spread across separate DAOs which prevented us from wrapping them in a single transaction. This ensures that all are committed, or none are.